### ISSUE TITLE
Unable to register: verification code not received

### ISSUE THREAD

**Issue by sivaraam** [signalapp/Signal-Android#7980]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
I'm unable to register my Indian mobile number to open an account i n the Signal app. I don't receive a verification code. I even tried to obtain the verification code via Call, and didn't receive a call either.

### Steps to reproduce
- Install the app
- Enter an Indian mobile number for registration
- Tap on Register

**Actual result:** 
I'm not able to get the verification code via SMS or Call.

**Expected result:** 
I get a verification code.

### Device info

**Device:** Samsung Galaxy j1 ace
**Android version:** 5.1.1
**Signal version:** 4.23.4

**Comment by sivaraam** [comment]
OK, I got the verification call, at last after about 1.5 hours. It wasn't a nice experience as it woke me up when I was sleeping.

It would be better to avoid making a call after a particular time limit.

**Comment by vishal-android-freak** [comment]
I am not able to register either via sms or call. Not receiving any of them.

**Comment by masonlouchart** [comment]
I always receive _"Unable to connect to service. Please check network connection and retry."_ message. It's displayed super quickly so I suppose the app did not even tried to reach the server.

**Comment by vitoreiji** [comment]
I have also not received the verification code via SMS and no call came through. I started this process over nine hours ago, and still no success.

Intermittently when trying I will get the message "Unable to connect to service. Please check network connection and retry." as described by @masonlouchart, but not every time. 

My country code is +255 (Tanzania) and my number has 9 digits.

My device is a Motorola Moto G4 Plus and I have Lineage OS installed (14.1-20180725-NIGHTLY-athene). I have installed the latest Signal app version (4.24.8) from Google Play.

Related: #5384 #6027

**Comment by Santihago** [comment]
> Also not able to register either via sms or call. Not receiving any of them. Not even after several (5 now) days.
> Country: Belgium (+32)
> my number has 10 digits like: 0478 xxx xxx
> 0 is automatically removed if I enter it.
> 
> additional info:
> Latest Signal app from Google Play Store
> Phone: Samsung Galaxy Note 3, Android 5.0
> Carrier/provider is Proximus

Same problem as verstken, from Belgium. Not receiving sms or calls. 
Number format as indicated (+32 470 12 34 56)
Phone:  iOS 12, iPhone 6s
Version: latest Signal app from App Store
Carrier: Mobile Vikings.

**Comment by greyson-signal** [comment]
@five-c-d did some investigation into a similar issue and wrote about his findings here:  https://github.com/signalapp/Signal-Android/issues/8148#issuecomment-426463464. Can you check that issue and see if you might have a similar problem?

**Comment by five-c-d** [comment]
@Santihago it looks like <a href="https://corporate.orange.be/en/news-medias/orange-belgium-and-medialaan-sign-full-mvno-agreement-welcomes-jim-mobile-and-mobile">MobileVikings is an MVNO</a> of the <a href="https://mobilevikings.be/en/mobile-vikings/news/mobile-vikings-chooses-the-orange-network/">OrangeBelgium network nowadays</a>, so you **could** be experiencing the same STOP-of-cmTelecom-also-inadvertently-blocks-Twilio behavior in #8148 ... 

However from what I can tel Proximus is NOT on the same backend as OrangeBelgium, they are a <a href="https://en.wikipedia.org/wiki/Proximus">hardware-independent competitor</a>.  There were a few internet-hits linking Proximus and CM Telecom, but they were in Dutch so I'm not sure what they said  :-)  It is definitely the case that Proximus is connected to cmTelecom however == https://help.cmtelecom.com/en/text/countries/sending-sms-to-belgium ... "CM has a direct route in Belgium and direct connections to all Belgian operators. We have direct access to a live MNP database ['mobile number provider'?] in Belgium... The Belgian Blacklist Mechanism requires companies to handle [and] immediately process STOP opt-out messages sent by consumers. ...Operators:  Proximus, OrangeBelgium, Base, Telenet, Lycamobile, NMBS, MundioMobile, ElephantTalk, Voxbone, JoinExperience, Unleashed" (these are all GSM services)

Now, if the mis-configuration is on the CM Telecom backend (as opposed to OrangeBelgium backend only), the STOP-difficulty might inadvertently impact *all* the carriers above... but only for endusers who have at some point previously replied STOP to a cmTelecom-generated message, and only if signal-servers use twilio and in turn twilio uses cmTelecom in Belgium?  So it is not fully clear to me what is going wrong yet, or if all these Belgium-issues are indeed related to each other (rather than just coincidence). 

If possible, can folks that are having no-SMS-and-no-robocall trouble please capture some debuglogs and paste the URL here?  As I understand it, the procedure is as follows:  
*  download signalapp from the play store 
*  install it onto your smartphone
*  during registration, you can get access to a debug log by the following method:  
*  tapping eight times upon the phone-icon (floating action-button) in the middle of the screen
*  when prompted, generate a debug-log, review it, and submit the URL
*  paste the generated URL here into github for folks to look over
*  this debug-log-capture <a href="https://github.com/signalapp/Signal-Android/blob/4.26/src/org/thoughtcrime/securesms/RegistrationActivity.java#L246">should work with signal4android 4.26</a> and higher, which I believe is fully rolled to to all playStore downloads now... however, not sure if the feature exists in signal4ios yet?  

And maybe @greyson-signal can confirm whether or not signal-server sends auto-generated SMS commands to Twilio... and in turn Twilio subcontracts out to CM Telecom for local SMS gateway services in Belgium?

**Comment by Undeadhunter** [comment]
Same issue here also using mobile Vikings. 

From what I understand is that they updated their backend recently 

Debug log 
https://debuglogs.org/dffffdd07766d1da079b9740f62f2582d15571f60246c46fba10cb3a7d7f188b

**Comment by Wouterdek** [comment]
I've been having the same problem for the past week. No verification SMS, nor calls. Tried on 2 different phones, but I could not verify on either. My carrier is also Mobile Vikings.

[Debug log](https://debuglogs.org/64c4bf2b16f3506b7cedea62ca4de8e25652e069119bfaa60153a298c54d65f2)

**Comment by letkan** [comment]
this happened to me today, no sms, no call received. Signal 4.29.4 built locally.

**Comment by verstken** [comment]
Is there someone in this thread who can enlighten us (non-developers) what's going on?
It seems like this thing isn't app related but more server-related... or maybe blocking of sms and calls globally, like some kind of conspiracy.

No, serious.. is there any lead where this comes from and when this could be resolved? I already had to tell three people they can't register when they wanted to try Signal after i've been telling them to use it for years.

Thanks

**Comment by letkan** [comment]
Agreed, this is embarrassing. Already having a hard time convincing people to use this instead of that spyware/crime_promoter. At least, a more descriptive error in case the server is down instead of "Unable to connect to service. Please check network connection and try again." which was thrown to me the other day after a few tries to register although my network connection was just fine. Maybe a registration by email would be more reliable?

**Comment by Wouterdek** [comment]
It would also be great if it was possible to transfer an account from one phone to another, or maybe add a phone like you add a desktop app installation. After all, if I already verified my phone number once, it shouldn't be necessary to do it again if I just want to reuse the same account on a different device, right?

**Comment by five-c-d** [comment]
I can attempt to shed a small amount of light:  first, what is going on?  

1.  with issue#7980, there is an (unexplained) delay when receiving the optional robocall, sometimes of many hours.  This happened to sivaraam (unspecified India telco) w/ 90 minute delay back in July 2018, vitoreiji (unspecified Tanzania +255 telco), letkan (unspecified carrier and unspecified country) many hours later, and over in the forums at least one other person MarcSN311 (unspecified Germany +49 telco) many hours later and intermittent.  https://community.signalusers.org/t/call-verification-takes-forever/4479  If you have debuglogs taken from signal4android during your failed registration-attempt, and you intermittently receive SMS or robocalls but sometimes only hours later, please post a comment to 7980 specifying your country-code, your carrier, whether you transferred your number, whether you were roaming, and any other pertinent details which might cause you to fail to receive the SMS or the robocall.  The signal-server uses a 3rd-party backend gateway for actually generating SMS-and-robocalls called Twilio.  The delay might be caused by a bug in signal-server (very unlikely), by a misconfiguration or problem with Twilio (has happened a couple times in past nine years), or by a carrier-and-country-specific problem (most likely).  Please use the forum-thread for non-github discussion of this kind of thing:  https://community.signalusers.org/t/call-verification-takes-forever/4479 , which I will try to help with as well. 

2.  there is a completely separate and unrelated issue#8148, that is specific to the Benelux area and carriers OrangeBelgium, their MVNOs such as MobileVikings, and various other EU carriers which have some relationship with CM Telecom (a competitor to Twilio that does SMS-gateway text-blasts in the EU).  In at least one case, covered in 8148, sending a 'STOP' message to a particular non-signal-related text-message from a CM Telecom client caused *all* incoming calls and texts to be halted, including those from signal-server's backend SMS gateway Twilio -- unclear if this is because Twilio subcontracts to CM, or there was a bug in OrangeBelgium config, or what exactly.  It is *possible* but not guaranteed to be the case, that the problems experience by Wouterdek (Mobile Vikings), Undeadhunter (MobileVikings), and Santihago (MobileVikings), plus possibly verstken (Proximus which is independent of OrangeBelgium but *does* have a backend-relationship to CM Telecom of some sort) are related to issue#8148.  Can any of you confirm whether you have ever replied 'STOP' to any telco-spam or service-messages or similar?  If you call your carriers, and try registering for signalapp whilst on the phone with their tech-support, can they see the SMS messages arriving, that you never get on your devices?  Forum-thread for these Benelux-specific regional errors is here == https://community.signalusers.org/t/blacklist-of-signal/ , and I would *very* much like to get to the bottom of this one.  

3.  there are actually several localized conspiracy-theories happening, but they are generally country-specific and sometimes carrier-specific, which includes censorship within a handful of nations (usually by the government which has direct control over the nationalized telecom infrastructure), so if you are in one of those countries you will experience not merely the inability to receive sms-and-robocall registrations, but the inability to use signalapp (unless you manage a Tor connection or VPN connection perhaps) as the signal-servers themselves will also be getting censored.  You can find your county here, https://github.com/search?q=org%3Asignalapp+censorship&type=Issues , and you can open a new forum-thread please, somebody will try to help you out.  These are quite hard to solve, of course, but the signalapp devs and the forum-participants DO want to see them fixed.  See also, https://github.com/search?q=org%3Asignalapp+domain+fronting&type=Issues , one of the main forum-threads is https://community.signalusers.org/t/censorship-how-does-registration-work-in-censored-countries/2618 

4.  If you try and fail to register "a bunch of times" then you will start getting HTTP 413 errors, which is a rate limiting throttle.  In your debuglogs you will see "RateLimitException" which is shorthand for "please stop trying and wait until tomorrow" because signal-server has decided you have exceeded the allowed number of registration-attempts.  Be patient, wait 24 hours, make sure you have very solid 'five bars' visibility with the cellular network and non-flaky internet connectivity as well, and then try slowly and deliberately one more time.  Hammering on the register-now button in frustration is not the answer (unfortunately)

5.  These are not the only possibilities, they are just the ones I am personally familiar with.  International telephony integration is a huge pain, and signalapp is not the only software suffering from it; be patient, and with gumption and persistence you can often find a way to solve the difficulty in your specific circumstances.  Open a forum-thread if you think your situation is not covered, or if you want to ask for help with trying to troubleshoot your specific difficulties == https://community.SignalUsers.org 

Now, given that backdrop, what is the best thing you can do?  Well, that depends on your situation.  Generally speaking, the problem is almost always related to the carrier, or in a narrower technical sense, to the simcard, when sms and robocalls fail to arrive.  

*  Obviously, make sure you have 'five bars' (a clear interconnect to the nearest cell tower) and plenty of battlife, plus a very stable internet connection as well (wifi or dataplan -- try both methods) 
*  More pertinently, you can try travelling to a different network where your phone is roaming, make sure you are not disabling the dataplan/sms/etc in that state, and then try registering again.  
*  You can try using a different simcard in your same device, preferably from a different carrier, to rule out any device-specific issues.  
*  You can try putting your simcard with your number into a different model of smartphone with a stock OS, to remove any software-issues related to GooglePlay and GCM and association with a gmail username ans such from the equation.  
*  You can call your carrier tech-support, and then try registering to receive the SMS conf-code during the phone call -- it is unencrypted so they can often monitor it, and will be able to read you the 6-digit conf-code verbally, allowing you to register (that trick "solved" #8148).  
*  You can, if your personal finances allow, change to a new carrier-plan, or a new carrier, or a new phone, or a new phone-number, or perhaps all of the above simultaneously.  

Other people in the forum might offer other advice, but those are the steps that seem to be the most fruitful.  While there, you can also help with various forum-threads related to some feature-requests that would (once implemented) vastly improve the registration process.  For instance, there has been significant design-progress at https://community.signalusers.org/t/registering-with-an-email-address/919 which would possibly help mitigate the registration problems.  My own suggestion is https://community.signalusers.org/t/feature-request-registration-via-calling-in-as-alternative-to-receiving-robocall-or-sms/4269 which was promptly shot down ;-)   Wouterdek's suggestion for multiple device linkage is mentioned https://community.signalusers.org/t/multiple-seperate-user-profiles-for-example-but-not-exclusive-to-dual-sim-phones/2361 as well as https://community.signalusers.org/t/multiple-phone-numbers/4011 but the reason registration is still needed each time is for security, otherwise anybody with your phone number could 'start using it on a new device' for you.  

Appreciate you folks trying to help with a maddening bug, it is no fun troubleshooting telco problems.  Phone numbers should "just work" is what people tend to think, but the reality is they are awful  :-)

**Comment by letkan** [comment]
This "explanation" is mind bugling, take the sim card out, use the phone in roaming, say a prayer, buy a new phone, kill a black cat, who has the time or more importantly, the motivation to go through all these? If the method of registration, of sending an sms or making a call is so flawed due to all these intermediaries as you describe, why was it not changed all these years? In other words, if one manages to register in one minute or so, it's a matter of luck, from your perspective, which is why I don't buy a word of it. It's a simple thing, register a device with a service which is working poorly and must be fixed or people will stop using the service.

**Comment by five-c-d** [comment]
@letkan, what is the country-code of the phone number where you are unsuccessfully attempting to register from?  Which carrier are you using, for the SMS/telephony service on your smartphone?  What operating system are you using, is it a stock ROM provided by the vendor?  Any other information you can provide, that will help troubleshoot your specific circumstances?

**Comment by Wouterdek** [comment]
> Can any of you confirm whether you have ever replied 'STOP' to any telco-spam or service-messages or similar?

I definitely never have done anything like this. I also just tried using [this support article](https://support.twilio.com/hc/en-us/articles/223134027-Twilio-support-for-opt-out-keywords-SMS-STOP-filtering-) to see if I could 'unblock' the service by sending UNSTOP to 8850 or 8810. Still doesn't work unfortunately.

> If you call your carriers, and try registering for signalapp whilst on the phone with their tech-support, can they see the SMS messages arriving, that you never get on your devices?  Forum-thread for these Benelux-specific regional errors is here == https://community.signalusers.org/t/blacklist-of-signal/ , and I would _very_ much like to get to the bottom of this one.

I'll see if I can call Mobile Vikings support tomorrow.

> Appreciate you folks trying to help with a maddening bug, it is no fun troubleshooting telco problems. Phone numbers should "just work" is what people tend to think, but the reality is they are awful :-)

I understand that this bug is not really on your end which makes this hard to diagnose and resolve. It shows that telco can be pretty unreliable and the system really needs alternative authentication mechanisms to prevent this single point of failure. I very much appreciate the effort you, and the rest of the Signal team, are making to improve this app.

**Comment by Undeadhunter** [comment]
I've got a ticket open with MV already since 3 weeks or so, it is definitely on their side. Talked with the support guy for a while and I am not the only one who this is happening to.

I've had issues with any type of SMS system (few exceptions). It is most likely related to their SMS spam filter on their new system.
Paypal, Signal, Waze to name a few. 

I am going to assume the people here (like me) have migrated to their new system (you should have replaced your SIM) and are having issues since. 

All Signal devs can do is wait until MV fixes their stuff this does not seem to be a Signal bug/issue.

However this has been going on for over a month now, so I'm looking into switching providers if it goes on for much longer.

**Comment by Notatechdud3** [comment]
> This "explanation" is mind bugling, take the sim card out, use the phone in roaming, say a prayer, buy a new phone, kill a black cat, who has the time or more importantly, the motivation to go through all these? If the method of registration, of sending an sms or making a call is so flawed due to all these intermediaries as you describe, why was it not changed all these years? In other words, if one manages to register in one minute or so, it's a matter of luck, from your perspective, which is why I don't buy a word of it. It's a simple thing, register a device with a service which is working poorly and must be fixed or people will stop using the service.

Letkan, I had somewhat the issue as you and it was five-c-d who dragged me through the entire proces. So please don't flame him for giving you a clear explanation (and not an "explanation") trying to resolve **your issue, not his**. Never bite the hand that feeds you ;-)

> > Can any of you confirm whether you have ever replied 'STOP' to any telco-spam or service-messages or similar?
> 
> I definitely never have done anything like this. I also just tried using [this support article](https://support.twilio.com/hc/en-us/articles/223134027-Twilio-support-for-opt-out-keywords-SMS-STOP-filtering-) to see if I could 'unblock' the service by sending UNSTOP to 8850 or 8810. Still doesn't work unfortunately.
> 
> > If you call your carriers, and try registering for signalapp whilst on the phone with their tech-support, can they see the SMS messages arriving, that you never get on your devices?  Forum-thread for these Benelux-specific regional errors is here == https://community.signalusers.org/t/blacklist-of-signal/ , and I would _very_ much like to get to the bottom of this one.
> 
> I'll see if I can call Mobile Vikings support tomorrow.
> 
> > Appreciate you folks trying to help with a maddening bug, it is no fun troubleshooting telco problems. Phone numbers should "just work" is what people tend to think, but the reality is they are awful :-)
> 
> I understand that this bug is not really on your end which makes this hard to diagnose and resolve. It shows that telco can be pretty unreliable and the system really needs alternative authentication mechanisms to prevent this single point of failure. I very much appreciate the effort you, and the rest of the Signal team, are making to improve this app.

Wouterdek, I think your issue is not related to MobileVikings but more to a mysterious company called CM (nee, niet die cm, de andere cm). I assume that somewhere, sometime you messaged STOP to a certain number because you got annoyed by the messages companies were sending. This means you blocked a lot of services without knowing it and you need to mail CM at cmsupport@cm.nl with your phonenumber and ask them to have you unblocked for commercial purposes. Keep me updated please.

**Comment by letkan** [comment]
> Letkan, I had somewhat the issue as you and it was five-c-d who dragged me through the entire proces. So please don't flame him for giving you a clear explanation (and not an "explanation") trying to resolve your issue, not his. Never bite the hand that feeds you ;-)

Well, maybe I was a little harsh, my main point was/is that blaming providers, figuring out all kinds of workarounds, waiting for a day or two are not the solution to the problem.  Reality is that registering to Signal service the way it is done now is a hit and miss and must be changed.

**Comment by sivaraam** [comment]
> This happened to sivaraam (unspecified India telco)

FWIW, the telon operator I faced an issue with is [airtel](https://www.airtel.in/). It has since been resolved though. It was just a delay. I was able to successfully register some time after I initially faced the issue.

**Comment by five-c-d** [comment]
sivaraam, thanks, and yes there are known problems (albeit with unexplained root causes that I am aware) with some telcos in India, per 'unofficial' wiki post here, https://github.com/signalapp/Signal-iOS/wiki/SMS-Verification-problems---Network-issues , though in official-but-closed-by-a-bot #6270 there is no mention of AirTel nor India 

@letkan , I for one heartily && definitely agree with your main point, the question is just, HOW to change in a way that has more upsides than downsides.  See forum-threads I linked to above; perhaps one of us here can come up with a proper non-bandaid solution.   

@wouterkdek , you have no obvious failures near the end <a href="view-source:https://debuglogs.org/64c4bf2b16f3506b7cedea62ca4de8e25652e069119bfaa60153a298c54d65f2 
">of your debuglog</a>, but there is a "rare" error message a couple days earlier than the time you submitted the file:  

```
2018-10-13 19:45:46.967 GMT+02:00 I PlayServicesUtil: Play Services: 0
2018-10-13 19:45:46.968 GMT+02:00 I TextSecurePreferences: Setting push registered: false
2018-10-13 19:45:47.142 GMT+02:00 W PushServiceSocket: Push service URL: https://textsecure-service.whispersystems.org
2018-10-13 19:45:47.144 GMT+02:00 W PushServiceSocket: Opening URL: https://textsecure-service.whispersystems.org/v1/accounts/sms/code/+*********38
2018-10-13 19:46:09.581 GMT+02:00 W PushServiceSocket: Push service URL: https://textsecure-service.whispersystems.org
2018-10-13 19:46:09.583 GMT+02:00 W PushServiceSocket: Opening URL: https://textsecure-service.whispersystems.org/v1/accounts/voice/code/+*********38
2018-10-13 19:46:34.762 GMT+02:00 W PushServiceSocket: Push service URL: https://textsecure-service.whispersystems.org
2018-10-13 19:46:34.764 GMT+02:00 W PushServiceSocket: Opening URL: https://textsecure-service.whispersystems.org/v1/accounts/code/111234
2018-10-13 19:46:34.896 GMT+02:00 W RegistrationActivity: null
2018-10-13 19:46:34.896 GMT+02:00 W RegistrationActivity: org.whispersystems.signalservice.api.push.exceptions.AuthorizationFailedException: Authorization failed!
```

Wouterdek, do you remember if you tried manually entering 111-234 as your confirmation-code at some point on the 13th?  Or did the signal4android client auto-generate that URL with 111234 itself, because you told it to integrate with SMS and automagically handle the replies?  

My understanding is that under normal conditions the HTTP 403-or-401 error should 'never' occur unless there is a serious problem with disk corruption or keystore damage or similar.  I saw the same intermittent failure in #8250 which is a Belarus-specific problem with never receiving the sms-nor-robocall, see comment over there.  Have you tried installing any non-stock OSes or flashing your LG's ROMs or anything like that?  It looks like you are running the normal LG vanilla ROM with android 6 at the moment, but what about in the past?

**Comment by sivaraam** [comment]
>  sending a 'STOP' message to a particular non-signal-related text-message from a CM Telecom client caused all incoming calls and texts to be halted, including those from signal-server's backend SMS gateway Twilio

I guess I now understand why I didn't get the text message for verifying my number in Signal. I have sent what you call the STOP message to a number (1909) to stop receiving promotional calls and messages. They call it the Do Not Disturb (DND) registry here. I didn't know it causes issues like these. I received the verification _call_, though.

> International telephony integration is a huge pain, ...

Interesting I've never heard of issues like these from people who use WhatsApp. Just wondering what they might be doing to avoid this, at least in our country (India)?

>  though in official-but-closed-by-a-bot #6270 there is no mention of AirTel nor India

I'm not sure how #6270 is related here. It seems about an issue about importing the message database which seems unrelated to this issue.

**Comment by Wouterdek** [comment]
> Wouterdek, I think your issue is not related to MobileVikings but more to a mysterious company called CM (nee, niet die cm, de andere cm). I assume that somewhere, sometime you messaged STOP to a certain number because you got annoyed by the messages companies were sending. This means you blocked a lot of services without knowing it and you need to mail CM at [cmsupport@cm.nl](mailto:cmsupport@cm.nl) with your phonenumber and ask them to have you unblocked for commercial purposes. Keep me updated please.

I'm certain I did not. And even if I did, I tried sending an unblocking message which had no effect. I'm also not receiving any robocalls, are those handled by twilio/cm-telecom aswell? However, as Undeadhunter suggested, I have migrated to the new system used by MV so the problem is probably related to that. What's weird is that SMS-verification for other services like Github, Telegram, Microsoft and Dropbox do still work.

> Wouterdek, do you remember if you tried manually entering 111-234 as your confirmation-code at some point on the 13th? Or did the signal4android client auto-generate that URL with 111234 itself, because you told it to integrate with SMS and automagically handle the replies?

I probably entered that manually after a few attempts to see what would happen.

**Comment by Notatechdud3** [comment]
> I'm certain I did not. And even if I did, I tried sending an unblocking message which had no effect. I'm also not receiving any robocalls, are those handled by twilio/cm-telecom aswell? However, as Undeadhunter suggested, I have migrated to the new system used by MV so the problem is probably related to that. What's weird is that SMS-verification for other services like Github, Telegram, Microsoft and Dropbox do still work.

Have you tried sending them a mail? Yes or no? And when did you send the unblock message? Did you get any reply?

**Comment by five-c-d** [comment]
@sivaraam , I meant #6027 (fixed now thanks).  In your case the reason the robo-SMS was not delivered via Airtel within India, is most likely *unrelated* to #8148 and 'STOP' misconfiguration, which is a specific carrier-bug in the Belgium/Netherlands/Luxembourg region that impacts at least OrangeBelgium, and possibly other carriers.  (There is a *problem* with Airtel refusing to deliver SMS according to a wikipage but that wikipage gives no details so the reason SMS is failing there is unknown to me.)  There is not an open 'official' github issue for that, but #6027 is the place those kinds of problems used to be tracked.  As for the 90-minute delay, that is this #7980 report... it should not take that long.  

From @Wouterdek's latest message about his MobileVikings-the-OrangeBelgium-MVNO in Belgium, he is almost certainly *not* suffering from #8148 nor from #7980 either (the email-check suggested by Notatechdud3 will tell us more), but a distinct problem... which we likely ought to move into a new issue-number once confirmed... very likely related to Undeadhunter's comment about <a href="https://github.com/signalapp/Signal-Android/issues/7980#issuecomment-432940022">MobileVikings's new SMS-spam-filter</a>, which in addition to signalapp SMS is also blocking paypal and waze and so on.  

> What's weird is that SMS-verification for other services like Github, Telegram, Microsoft and Dropbox do still work.

> Interesting I've never heard of issues like these from people who use WhatsApp. Just wondering what they might be doing to avoid this

Signalapp currently has under 10m endusers (by my own guesstimation only... I have no inside info).  Telegram has >100m, and WhatsApp has >1000m, per internet estimates made by others.  If there is a specific carrier-misconfiguration in a specific country that breaks Telegram or breaks WhatsApp, the carrier gets an earful of complaining from ten times or a hundred times as many cellular-customers, as when there is a breakage in signalapp -- simply because there are fewer endusers of signalapp presently.  

p.s.  In the long run, methinks the real fix for signalapp being able to troubleshoot these kinds of problem pro-actively (rather than retroactively AFTER enough tech-savvy endusers get annoyed enough to start reporting troubles via github) is likely going to involve creating an automated test-farm of SMS-capable devices, and registering the test-farm devices with all the major carriers worldwide... then setting them in a loop to unregister and re-register themselves every N minutes.  

This is an expensive proposition: you need devOps/SQA people to maintain the test-farm, you need carrier-plans for all devices, you need to pay twilio (and in many countries also the carrier) for every robo-SMS you send ... which can get expensive where N is a small integer.  But the smaller you make N, and the more device-models and carriers you cover, the faster you can know about -- and take actions to correct -- any kind of misconfigurations/etc between the signal-servers and the signal4smartphone devices which are impacting the registration-pathway via the cellular-network.   

In the medium-run, maybe there is a cheaper fix:  a feature-request to implement a "re-registration dry-run" option that endusers can optionally enable on their own devices, which will periodically go through the motions of re-registering their phone-number, but WITHOUT unregistering them if the dry-run re-registration happens to fail.  https://community.signalusers.org/t/feature-request-ability-to-perform-a-re-registration-dry-run/4583

In the short-run, volunteer troubleshooters like yourselves and like myself will tend to carry the bulk of the burden:  calling carrier tech-support, swapping simcards amongst devices to rule out the problems, swinging a black cat around whilst trying from a roaming-connection or from a wifi-boosted-calling configuration, and so on.  This is a pain, for sure, but the upside is that the harder we work now, the less we will have to work in the long run:  as more and more millions of people are able to register successfully for signalapp via SMS-or-robocall, fewer and fewer carriers will get away with misconfigurations that break Signalapp for all those people (cf Telegram and WhatsApp with their 10x and 100x larger userbases).

**Comment by sivaraam** [comment]
> (There is a problem with Airtel refusing to deliver SMS according to a wikipage but that wikipage gives no details so the reason SMS is failing there is unknown to me.)

My suspicion is that the person who added it to the wiki page might have also faced the issue as he has opted-in to the Do Not Disturb registry. Also the [linked Twillio support page](https://support.twilio.com/hc/en-us/articles/223134167-Limitations-sending-SMS-messages-to-Indian-mobile-devices) also states about the STOP message issue *in India*.

**Comment by Wouterdek** [comment]
I was just about to call Mobile Vikings tech support, but I decided to just try one last time to do the SMS verification.

It worked. For some reason, the issue has been resolved for me. Maybe MV changed something in their back-end. Maybe the unblock message had an effect after a couple days. (Never got any sort of reply though) Whatever it was, it is no longer an issue for me.

**Comment by five-c-d** [comment]
@Wouterdek, can you please capture a debuglog, for upload here?  Maybe a compare and contrast with your previous (unsuccessful) one will help.  home > signal > threeDotMenu > settings > advanced > debuglog      Glad to hear it is functional for you, that is an improvement ... although we don't know exactly why it was functional.  Did you receive the SMS-conf-code, or did you have to tap Call Me and get the robocall?

**Comment by Santihago** [comment]
Now it worked for me too. I got my sms after 5 seconds (from an American +1 number) and auto-filled in. Problem solved.
iOS, Mobile Vikings.

**Comment by five-c-d** [comment]
@Wouterdek, when you uploaded a debuglog previously, you were running 4.26.2 which is signal4android stable, but this time your debuglog says 4.29.7 which I believe means you are on the beta-channel now.  Here are 3 recent attempts, with the first one involving hand-entering a guessed conf-code, the second one skipping that, and the third one succeeding.  

* fail == `2018-10-21 17:52:46.388 GMT+02:00 W PushServiceSocket: Opening URL: https://textsecure-service.whispersystems.org/v1/accounts/sms/code/+*********38`
* guess== `2018-10-21 17:53:10.141 GMT+02:00 W PushServiceSocket: Opening URL: https://textsecure-service.whispersystems.org/v1/accounts/code/096385`
* wrong== `2018-10-21 17:53:10.291 GMT+02:00 W RegistrationActivity: org.whispersystems.signalservice.api.push.exceptions.AuthorizationFailedException: Authorization failed!`<br /><br />
* bump == `2018-10-24 23:07:03.566 GMT+02:00 I DatabaseUpgradeActivity: LastSeenVersion: 407`
* fail == `2018-10-24 23:07:10.429 GMT+02:00 W PushServiceSocket: Opening URL: https://textsecure-service.whispersystems.org/v1/accounts/sms/code/+*********38`<br /><br />
* bump == `2018-10-27 13:04:14.878 GMT+02:00 I DatabaseUpgradeActivity: LastSeenVersion: 419`
* work == `2018-10-27 13:07:35.774 GMT+02:00 W PushServiceSocket: Opening URL: https://textsecure-service.whispersystems.org/v1/accounts/sms/code/+*********38`
* sms! == `2018-10-27 13:07:40.900 GMT+02:00 I SMSListener: Got SMS broadcast...`
* conf == `2018-10-27 13:07:41.029 GMT+02:00 W` <a href="https://github.com/signalapp/Signal-Android/blob/master/src/org/thoughtcrime/securesms/service/SmsListener.java#L149">`SmsListener: Got challenge!`</a> 
* auto == `2018-10-27 13:07:41.040 GMT+02:00 I RegistrationActivity: Got a challenge broadcast...`
* auto == `2018-10-27 13:07:42.354 GMT+02:00 W PushServiceSocket: Opening URL: https://textsecure-service.whispersystems.org/v1/accounts/code/194299`
* okay == `2018-10-27 13:07:46.149 GMT+02:00 W PushServiceSocket: Opening URL: https://textsecure-service.whispersystems.org/v2/keys/`

@greyson-signal, can you peek at the Twilio and signal-server logs when you have time please, and see whether those three events above, show anything on the backend that looks suspicious/etc?  If nothing shows there, then we can assume that MobileVikings altered their antispam-sms-filter, or that the 'unstop' requests Wouter sent previously were a success, or *maybe* that the 4.29.7 upgrade helped somehow.  

@Undeadhunter, as the third MobileVikings person, with luck you will also now have success: 

1.  try to register with your current version of signalapp, should be 4.26.2 methinks 
2.  wait a few hours, then try again using 4.29.7 **if you are willing** -- however note that the beta channel is NOT as "safe" as the stable channel, if you are not interested in this approach that is no problem.  Recommend you enable the backup-feature in your signalapp settings, write down the backup-decrypt-passphrase, make sure you get a full backup-file created, and thn offload that backup of your signal-conversations/groups/etc to another device prior to attempting the beta-channel, in any case.  Skip to step #4 if you would rather stay on stable
3.  give it another couple hours to make sure your database-migration-upgrades are complete (leave signal4android open), then give 4.29.7 another try 
4.  wait a few more hours, then try sending the messages that Wouterdek used from the Twilio helpdocs, https://github.com/signalapp/Signal-Android/issues/7980#issuecomment-432832260 , and try registering again 
5.  wait 24 hours then try again, then per Notatechdud3 and #8148 try emailing CM Telecom the local SMS gateway at cmsupport@cm.nl , with your cellular number, asking them to please have it unblocked for commercial purposes , then try again an hour later 
6.  wait a few hours and try again 

@Santihago, can you check your version of signal4ios please, home > signal > settingsMenu > about (which I believe is <a href="https://support.signal.org/hc/en-us/articles/360007059212-How-do-I-ensure-Signal-is-up-to-date-#update_ios ">the correct pathway for iOS</a> folks), to see if you are on the invited-beta version?

**Comment by ajzett** [comment]
> OK, I got the verification call, at last after about 1.5 hours. It wasn't a nice experience as it woke me up when I was sleeping.
> 
> It would be better to avoid making a call after a particular time limit.

I recently reset my device, and now it is doing the same thing. I tried to get it to call me last night, but it woke me up _12 hours later_ with two calls back-to-back. 

I'm on android custom ROM crDroid 8.1 with MindTheGApps and latest stable Play Store release of Signal. I've flashed ROMs before and I've never had an issue with re-registering. I thought I was going nuts, but it looks like others are having the same issue.

**Comment by five-c-d** [comment]
@Navneet-Suresh , there is a forum-thread where we are trying to troubleshoot issues with Airtel, since you are not alone experiencing problematic behavior (see signal4<a href="https://github.com/signalapp/Signal-iOS/issues/3929">ios #3929</a>).  If you can capture a debug-log, by tapping eight times on the floating phone-icon during registration and posting the URL here, that will help to verify you are not having an unrelated issue.  If you wish to help do some community-debugging of your device and carrier in the community forum-thread, you can login there with your github credentials == https://community.signalusers.org/t/registering-in-india-airtel/4684  

To narrow down exactly what the trouble is, it will help if some volunteers can pin down whether SMS works from airtel-to-airtel customers, whether SMS works from an international USA longcode to an airtel number, and ideally whether it is possible for an airtel customer to register their handset with a standalone custom signal-server outside of OWS control (to rule out censorship as opposed to gateway-misconfiguration).  You can direct-message me in the forum if you would like to assist in this fashion.  In the long run we will likely need assistance from an OpenWhisper developer with access to the backend Twilio logs, plus possibly tech support from Airtel representatives, to fully solve the difficulty.  

But there are a lot of Airtel customers and it would be great if signal4ios and signal4android could actually receive the SMS, and failing that, actually receive the robocall in a timely fashion rather than hours later.

**Comment by Navneet-Suresh** [comment]
@five-c-d I would like to help. I have taken a debug log. I am attaching it below.
https://debuglogs.org/f881d4009a46f4ae318f22f4becbe50d7562a8679b8412efbdd695038840c6f0
SMS verification works for other apps just fine. I am unable to receive verification codes on other networks too. (Vodafone and Reliance Jio).

**Comment by sivaraam** [comment]
Hi @Navneet-Suresh, it's great you're willing to help debugging this issue. Did you not receive the verification code via call either? I ask this as I'm also a airtel customer in India who also faced this issue. I *did not* get the verification code via SMS but I *did* get the verification code via call (though it was delayed for the first time).

**Comment by Navneet-Suresh** [comment]
I did recieve the call a few time delayed by few hours(4-8 hours delay) on my Airtel no. But I didn't receive the call on Vodafone or Jio. Today I tried to register again on Airtel its still not working and No call so far.
Here are the no's that I got call from.
+1 702-522-1202 (Twice) 3:23 PM on 02-11-2018
+1 530-994-5246 (Once) 11:08 AM yesterday.(04-11-2018)

**Comment by verstken** [comment]
Tried to register again yesterday with my phone number (+32) provider Proximus.
Registration was successful.
That makes two providers in Belgium that work again. (Mobile Vikings was mentioned here before to work successful again)

I haven't changed anything. Not a request to unblock or anything. So i guess the provider did something to unblock calls and sms from Signal.

**Comment by kalleyne** [comment]
On HMD Global Nokia 3.1 with Android 8.1.0.
Location: USA
Very persistent error message trying to Register with Signal 4.29.7.

"Unable to connect to service. Please check network connection and try again."

Due to all the strangeness trying to receive the verification code via SMS and via call, I decided to contact T-Mobile this morning to have them remove all SMS and adult blocks on my plan.

On this phone I've removed and reinstalled Signal through Google Store more than a dozen times today with reboots. Only once did it seem to work and I was able to obtain a correct verification code. But I had to manually type that in before the countdown timer expired.

It seems that I eventually get verification codes via SMS but often an hour or longer after I initially requested them. 

Here is my debug log from one of my many failed attempts:

[https://debuglogs.org/ef58ad50911b489b8d64283d76ecb137c1eb401f1e656d742e01c634a5c5ddb0](url)

**Comment by five-c-d** [comment]
@kalleyne , I cannot see any SMS being received in your debuglog, but I do see that you are getting an error-message which is the HTTP 413 rate limit exceeded response from signal-server.  If you try to register the same phone number more than about a dozen times, signal-server stops accepting requests (it costs money to generate robocalls and sms-blasts that Signal Foundation would prefer to spend elsewhere if possible).  So here is what you need to try:  

1.  do not try to re-register your +1xxxyyyzz70 phone number for a few hours... ideally wait 24 hours, though perhaps 8 hours will be enough.  
2.  after this time has passed, slowly and deliberately open up signalapp on your android device, enter your +1xxxyyyzz70 number, and click to receive the sms.  Wait patiently, at least two or three minutes, without doing anything else.  
3.  if the SMS arrives, jot down or memorize the 6-digit conf-code, and then tap it into signalapp manually.  (If you have enabled SMS integration permission then signalapp can automagically detect the incoming SMS, but if you denied that permission because you want to keep your encrypted messaging app separate from your unencrypted SMS app then you need to enter the conf-code by hand.)  
4.  once two or three minutes have gone by without the incoming SMS arriving, you can fallback on the robocall option by tapping Call Me Instead, then waiting patiently for the robocall to arrive.  It should not take 90 minutes, but it might take ten or fifteen.  
5.  when the robocall arrives, the automated voice will verbalize the conf-code, which you should tap into signalapp.  

You can re-try this procedure a few times in a row, but NOT dozens of times in a row, before signal-server will stop accepting requests.  You also need to wait a sufficient amount of time in between registration attempts -- a few seconds is not long enough, you end up stepping on the previous request-attempt.  

However, in such cases where you ARE going too quickly and the signal-server starts rejecting subsequent retries, you should see a message saying as much... but this is a known bug, there is no user-visible rate-limit message at present.  Here is the code in question, <a href="https://github.com/signalapp/Signal-Android/blob/4.29.7/src/org/thoughtcrime/securesms/RegistrationActivity.java#L649">RegistrationActivity.java#L649</a> , where result==3 corresponds to the RateLimitException on line 625 of that java code which is for the RegLock feature.  If you are ONLY seeing the 'unable to connect to service please check network connection' message, and not a 'too many attempts' sort of message, that means that <a href="https://github.com/signalapp/Signal-Android/pull/7872/commits/c93fea8bfdf3a3e0933702a764bc4e0d3f15c9ca#diff-38f6c8c2554571e84e923467552255baR505">this pull-request</a> has not yet been upmerged into the main signalapp codebase, and that #7819 is still open.  

As a workaround until 7819 is fixed, on signal4android during registration you can tap eight times on the floating phone-icon in the middle of the screen, which will generate a debuglog that you can view.  If you see these kinds of errors near the bottom (debuglog is sorted chronologically) then registration-attempts are being rate-limited by signal-server and you need to pause for a few hours: 

* ``2018-11-05 15:47:37.563 CST W PushServiceSocket: Opening URL: https://textsecure-service.whispersystems.org/v1/accounts/sms/code/+*********70``
* ``2018-11-05 15:48:50.577 CST W PushServiceSocket: Opening URL: https://textsecure-service.whispersystems.org/v1/accounts/voice/code/+*********70``
* ``2018-11-05 15:50:41.728 CST W PushServiceSocket: Opening URL: https://textsecure-service.whispersystems.org/v1/accounts/sms/code/+*********70``
* ``2018-11-05 15:50:42.011 CST W RegistrationActivity: org.whispersystems.signalservice.api.push.exceptions.RateLimitException: Rate limit exceeded: 413``

Your debuglog that you uploaded has dozens of those HTTP 413 result.  I did not see any SMSListener in the debuglog, which means either you have disabled SMS integration, or no SMS arrived?  (Make sure you can receive a normal SMS at the number you are registering with by testing with a friend.)  The speed of the re-registration attempts almost looks like something is automated on your android device... or maybe you just have very fast thumbs?  :-)   It looks like you have a stock Nokia 00WW_2_590 ROM installed for the phone, do you have any special developer-applications installed? 

Only other exception in the logfile was a couple of places where there were "``ClassNotFoundException: com.mediatek.cta.CtaHttp``" which I believe is related to transient network errors, <a href="https://stackoverflow.com/questions/52858130/flutter-app-crashes-when-internet-gets-disconnected-while-runnig-transactions">per stackoverflow thread</a>.  So it is not impossible that SOME of the 'unable to connect' errors you are seeing, are actual network-related hiccups.  But most of the trouble is 413 rate-limiting.

**Comment by sivaraam** [comment]
This seems to have been fixed at least in India. See [linked comment](https://community.signalusers.org/t/registering-in-india-airtel/4684/12?u=sivaraam) and the following ones.

**Comment by Freewind65** [comment]
Same issue bye too.

No verification code/call with Vodafone Germany

**Comment by Notatechdud3** [comment]
> Same issue bye too.
> 
> No verification code/call with Vodafone Germany

Hello

Could you tell us if you are using a landline or a cellphone (I assume cellphone because you mention the verification code)? Is this the first time that you are using Signal? Android or IOs?

Please perform the following steps: 
- Download and install Signal
- During the registration proces, access the debug log by performing the following steps:
-- Tap 8 times on the phone-icon in the middle of the screen, when promoted, generate a debug log, review it (watch that there are no any private details, e.g. cellphonenumber in it) and the submit the url here please

**Comment by greyson-signal** [comment]
I think @five-c-d has done some excellent research here and found the core of this problem. We're making continual improvements to SMS registration, but in the meantime follow their tips. Thanks!

**Comment by f4grx** [comment]
Got the issue with a french operator that usually has no problem.

Using latest signal in LineageOS 14, the 8-tap debug log showed an exception related to SMS capture:
API_NOT_AVAILABLE related to microg and some google shit. Why is that used in Signal? Why does signal fail to run when the sms capture is not working? The SMS should still be sent, received by the user, and the code typed manually. I guess some simple try { } catch() { } is missing somewhere :)

The log contained the URL used to send the SMS, I copied and pasted it in firefox *on another computer*, that gave me an SMS with a code, and that passed verification. Problem solved!

**Comment by Firobe** [comment]
@f4grx Same here, french operato, LineageOS 14. Thanks for the workaround !

**Comment by mrusme** [comment]
Not receiving any SMS/calls on T-Mobile Germany right now. Tried it for the past three hours. [Unregister](https://signal.org/signal/unregister/) SMS won't arrive either. Seems like SMS delivery is broken?

**Comment by five-c-d** [comment]
@mrusme , can you please comment further on what you experienced, over in the forum-thread?  https://community.signalusers.org/t/should-phone-number-and-ip-address-be-from-same-country-for-sms-registration/6614/4   What VPN are you utilizing, in particular... maybe there is a conflict with that VPN provider and the signalapp registration process?  If you can upload a debuglog URL from the handset in question, that might help track down this problem also.  Most nums in Germany "work fine" for 99% of endusers 99% of the time, but there are reports every few months of the robo-SMS taking hours to arrive.  I'd like to find the root cause, if you can help with debugging please?

**Comment by usershmusername** [comment]
Comment made from my phone so apologies for brevity. I have a OnePlus 5t, on Vodaphone in UK. I was unable to register by sms as per OP. Using NoRootFirewall app I was able to identify a system app "com.*" (there were several names in the firewall entry including OnePlus and Qualcomm) that triggered during sign-up. I allowed the traffic only from Signal and I instantly recieved a registration code. Should I don my tinfoil hat?

**Comment by saurabhgupta050890** [comment]
I have recently tried Signal and still not receiving SMS on India number (+91 ). I was able to receive call though

**Comment by gfcapalbo** [comment]
from italy +39 number. No message received, requested phone call twice. Nothing after 2 hours.
this failed attempt was made on first app launch after installation.

Shut down app, tried again after 8 hours. Worked immediately with sms confirmation.

Signal for android 4.78.5 on a samsung galaxy A40.

**Comment by MMrhassel** [comment]
Moroccan too +212

**Comment by slorquet** [comment]
looks normal at this time since many people are migrating from whatsapp

**Comment by balamacab** [comment]
Hello, from Colombia (+57) I try to register to signal but receive the message related to "RegistrationActivity_unable_to_connect_to_service" or the message related to "RegistrationActivity_error_connecting_to_service" after receiving the SMS and confirming in the app.

**Comment by Shrlck-X** [comment]
Hello all! 

I am not receiving the Telegram confirmation call for a landline number in Brazil (+55). I have tried many times, on different days, and using either the Android and Desktop apps. Unfortunately SMS code is not an option... :-(

**Comment by bluepengwing** [comment]
> I've been having the same problem for the past week. No verification SMS, nor calls. Tried on 2 different phones, but I could not verify on either. My carrier is also Mobile Vikings.
> 
> [Debug log](https://debuglogs.org/64c4bf2b16f3506b7cedea62ca4de8e25652e069119bfaa60153a298c54d65f2)

Hi! I am also having the same problem. Just that I am just experiencing it now and I am from the Philippines. It's already my day 5 of attempting to register and wait for the verification code that never went through. Then I tried to register my sister's number (from a different network provider) and the code came. I am now planning to have another sim card by that other network provider, just that this number is my main one and I'm really hoping to have it fixed so it wouldn't be a hassle for me and my contacts. Thank you so much for all the infos posted here. Also I already tried the instructions sent by the signal support, yet still the same.

**Comment by Leon-zp** [comment]
Good afternoon. I have the same problem, I do not receive an SMS during registration and I do not receive a phone call. I am in the city of Berdyansk. The mobile operator is Russian +7Telecom.
